---
App[] running_apps
App[] available_apps